float32 psi1
float32 psi2
float32 movement